### DISCUSSION TITLE
Is it supposed to work with USB-connected adapters?

### DISCUSSION THREAD

**Discussion by pegleGrot** [smx-smx/ASMTool#15] [Q&A]
Trying with:
AsmIo*.sys v1.0.1.0, dated 2015/1
asmiodll.dll v1.0.2.0, dated 2014/12

asmtool.exe just says:
```
Unloading ASM Driver...
Loading ASM Driver...
Failed to load ASM IO Driver
```

**Comment by c4801725870** [comment]
> Trying with: AsmIo*.sys v1.0.1.0, dated 2015/1 asmiodll.dll v1.0.2.0, dated 2014/12
> 
> asmtool.exe just says:
> 
> ```
> Unloading ASM Driver...
> Loading ASM Driver...
> Failed to load ASM IO Driver
> ```

which chip do you have?

**Comment by pegleGrot** [comment]
ASM1153E.

**Reply by c4801725870** [reply]
> ASM1153E.

your usb vid and pid are not matched. you would have to recompile with the match. See here https://github.com/smx-smx/ASMTool/blob/master/AsmTool/AsmDevice.cs

**Comment by pegleGrot** [comment]
Well, that error comes earlier, from:
https://github.com/smx-smx/ASMTool/blob/4d70d0da35eff91884a40708453bfb54d7a90daf/AsmTool/Program.cs#L25-L27

Which is just a direct call to a DLL function:
https://github.com/smx-smx/ASMTool/blob/4d70d0da35eff91884a40708453bfb54d7a90daf/AsmTool/WindowsAsmIO.cs#L27-L31

I'm guessing either there are different versions of that DLL with different compatibility, or that it doesn't support the specific chip, or how it's connected through USB.
